osquery/tables/system/darwin/os_version.mm

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <cerrno>
+#include <sys/utsname.h>
+
+#include <string>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/sql/sql.h>
+#include <osquery/utils/conversions/darwin/cfstring.h>
+#include <osquery/utils/conversions/split.h>
+
+#import <CoreFoundation/CoreFoundation.h>
+
+namespace osquery {
+namespace tables {
+
+// Type used for function pointers to:
+// * _CFCopySupplementalVersionDictionary
+// * _CFCopySystemVersionDictionary
+typedef CFDictionaryRef (*copy_version_t)();
+
+CFDictionaryRef fetchSupplementalVersionDict() {
+  // Create a URL ref for the private framework we need
+  CFURLRef bundle_url = CFURLCreateWithFileSystemPath(
```

Comment:
`CoreFoundation` is not a private framework, I haven't tried it yet, but I don't think we need this, CoreFoundation is already linked in. Have you tried it without it?

edit: ah nvm, I see `CFBundleGetFunctionPointerForName` requires a bundle reference